test/reactive-membrane.spec.ts

```diff
@@ -40,4 +40,35 @@ describe('constructor', () => {
         expect(wet !== regex1).toBe(true);
         expect(wet.source).toBe("fooBar");
     });
+
+    it('should support proxies for branded non-plain objects', () => {
+        class Foo {
+            _x = 0;
+            get x() {
+                return this._x;
+            }
+            set x(v) {
+                this._x = v;
+            }
+            doubleX() {
+                return this._x * 2;
+            }
+        }
+        const tag = Symbol.for('@@myMembraneTag');
+        const blue = new Foo();
+        Object.defineProperty(blue, tag, {});
```

Comment:
@stephenh this is the most important line of all... you must brand any object that you wish to wrap in an observable proxy. In this case I'm installing a symbol as non-configurable, non-writable. But you could do weakmaps, or do this brand in your super class constructor, or anywhere else that later on you can identify the object in the `valueIsObservable` hook.

Reply:
> brand in your super class constructor,

Just curious, but since you said "_super_ class constructor", would this not work within `Foo` constructor itself, i.e. if I knew all `Foo`s were okay to be proxied, could I do:

```
class Foo {
  constructor() {
    Object.defineProperty(this, tag, {});
  }
}
```

Or even:

```
class Foo {
  [tag]: {}
}
```

My guess is that is okay...

Reply:
Foo is the last constructor in the proto chain, so yes... that's what I meant by super class constructor. anyone who extends it, and call super... will get a branded `this` value back after calling super. In theory it doesn't need to be done there... but anywhere in the proto chain... but I think it is easier, safer to do it there.

As for the examples, yes, your first snipped works great, and yes, your second will probably work as well, but it is going to have two drawbacks that the first one doesn't:

a) it is enumerable (which means it pops for for-in and other enumerable driven inspections.
b) it is removable/mutable, etc.

Btw, to make it more equivalent to the previous one, it's value should be set to undefined, that's what the first example does... remember that the 3 argument of `Object.defineProperty` is a descriptor, not a value.